### PR TITLE
[BUILD] Accept path list in OPENTELEMETRY_EXTERNAL_COMPONENT_PATH

### PR DESCRIPTION
**Pull request by ThomsonTan**
Extend `OPENTELEMETRY_EXTERNAL_COMPONENT_PATH` to support path list will help integration with multiple external exporters.

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: cmake/opentelemetry-build-external-component.cmake [hunk 1]
```diff
@@ -13,13 +23,18 @@
 if(OPENTELEMETRY_EXTERNAL_COMPONENT_PATH)
   # Add custom component path to build tree and consolidate binary artifacts in
   # current project binary output directory.
-  add_subdirectory(${OPENTELEMETRY_EXTERNAL_COMPONENT_PATH}
-                   ${PROJECT_BINARY_DIR}/external)
+  foreach(DIR IN LISTS OPENTELEMETRY_EXTERNAL_COMPONENT_PATH)
+    get_directory_name_in_path(${DIR} EXTERNAL_EXPORTER_DIR_NAME)
+    add_subdirectory(${DIR} ${PROJECT_BINARY_DIR}/external/${EXTERNAL_EXPORTER_DIR_NAME})
```
Comment (lalitb) [inline cmake/opentelemetry-build-external-component.cmake:28]: nit - can there be a conflict  if the path list is (say) `/dir1/subdir1 /dir2/subdir1`.

Comment (owent) [reply cmake/opentelemetry-build-external-component.cmake:28]: cmake use `;` to split list item, not space.I think there will no conflict here.

**Review by lalitb — APPROVED** [review]
LGTM with nit suggestions

File: cmake/opentelemetry-build-external-component.cmake [hunk 2]
```diff
@@ -1,10 +1,20 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+function(get_directory_name_in_path PATH_VAR RESULT_VAR)
+  # get_filename_component does not work with paths ending in / or \, so remove it.
+  string(REGEX REPLACE "/$" "" PATH_TRIMMED "${PATH_VAR}")
+  string(REGEX REPLACE "\\$" "" PATH_TRIMMED "${PATH_VAR}")
```
[the remark this answers is not in the document]
Comment (ThomsonTan) [reply cmake/opentelemetry-build-external-component.cmake:7]: Thanks. Haven't tried the the regex, will try the pattern.

**Pull request merged**